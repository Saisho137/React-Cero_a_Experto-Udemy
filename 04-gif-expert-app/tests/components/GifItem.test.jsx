import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GitItem";

describe('Pruebas en <GifItem />', () => {
    const title = "Shinji";
    const url = "https://eva.com/eva.png";

    test('Debería hacer match con el snapshot ', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('Debería mostrar el titulo ', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('Debería mostrar la imagen con el url y alt indicado', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        /*
        //screen.debug();
        expect( screen.getByRole('img').alt).toBe(title);
        expect( screen.getByRole('img').src).toBe(url); 
        */
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });
});