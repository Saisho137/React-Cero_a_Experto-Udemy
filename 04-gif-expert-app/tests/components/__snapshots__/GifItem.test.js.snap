// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Deber√≠a hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Shinji"
      src="https://eva.com/eva.png"
    />
    <h4>
      Shinji
    </h4>
  </div>
</div>
`;
